src/mgmt/cli_mgmt.c

```diff
@@ -699,6 +718,7 @@ dc_mgmt_net_cfg(const char *name, crt_init_options_t *crt_info)
 
 cleanup:
 	if (rc) {
+		D_FREE(g_serv_ranks);
```

Comment:
where would this be freed in the non-error case?

Reply:
probably should be in dc_mgmt_fini()?

Reply:
actually NLT just failed because of that, so i think just freeing it in fini should resolve this.
```
{
   <insert_a_suppression_name_here>
   Memcheck:Leak
   match-leak-kinds: reachable
   fun:calloc
   fun:d_calloc
   fun:dc_mgmt_net_cfg
   fun:daos_init
   fun:_cgo_c695f4df3957_Cfunc_daos_init
   fun:runtime.asmcgocall.abi0
   obj:*
}
```

Reply:
Indeed, there is some place where I need to free the allocation.
I am going to fix this asap.

- [ ] Fix memory leak

Reply:
> Indeed, there is some place where I need to free the allocation. I am going to fix this asap.
> 
> * [x]  Fix memory leak

Fixed with commit bae48ea068e35dd279bc347ba7b18f2befb62aa2